webapp/src/main/java/com/box/l10n/mojito/service/tm/search/TextUnitSearcher.java

```diff
@@ -441,6 +441,10 @@ NativeCriteria getCriteriaForSearch(TextUnitSearcherParameters searchParameters)
       c.setOffset(searchParameters.getOffset());
     }
 
+    if (searchParameters.isOrdered()) {
```

Comment:
I figured that was the case, I guess the question would be that is the performance impact of ordering the return bigger than the perf impact we're experiencing at present with multiple imports of batches?

For short circuiting to work we'll need to ensure that the files we push to Smartling are consistent in order otherwise the returned translated files can potentially change on each download and reduce the effectiveness of the short-circuiting. Happy to discuss alternatives that might better provide us with relatively consistent batches.

Reply:
As discussed in team sync, alternative consistent hashing approach will be investigated in a separate task to avoid introducing too many file generation changes in this performance related PR.